src/diffusers/models/unet_2d_condition.py

```diff
@@ -266,17 +267,47 @@ def __init__(
         self.conv_act = nn.SiLU()
         self.conv_out = nn.Conv2d(block_out_channels[0], out_channels, kernel_size=3, padding=1)
 
-    def set_attn_processor(self, processor: AttnProcessor):
+        if attn_processor_cls is not None:
+            attn_processors_keys = self.attn_processors.keys()
+            self.set_attn_processor({k: attn_processor_cls() for k in attn_processors_keys})
+
+    @property
+    def attn_processors(self) -> Dict[str, AttnProcessor]:
         # set recursively
-        def fn_recursive_attn_processor(module: torch.nn.Module):
+        processors = {}
+
+        def fn_recursive_add_processors(name: str, module: torch.nn.Module, processors: Dict[str, AttnProcessor]):
             if hasattr(module, "set_processor"):
-                module.set_processor(processor)
+                processors[name] = module.processor
 
-            for child in module.children():
-                fn_recursive_attn_processor(child)
+            for sub_name, child in module.named_children():
+                fn_recursive_add_processors(f"{name}.{sub_name}", child, processors)
 
-        for module in self.children():
-            fn_recursive_attn_processor(module)
+            return processors
+
+        for name, module in self.named_children():
+            fn_recursive_add_processors(name, module, processors)
+
+        return processors
+
+    def set_attn_processor(self, processor: Union[AttnProcessor, Dict[str, AttnProcessor]]):
```

Comment:
Setting a dict of `processor` will allow for much more flexibility here.

Reply:
So the idea is to add different processors for down/mid/up blocks? Interesting! Maybe we need to add tests to check that we can set a single processor for all three, or different ones :)